tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -834,6 +834,16 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(FunctionID function
                       " name=", caller.type.name, ".", caller.name, "()");
     }
 
+    // In NETFx, NInject creates a temporary appdomain where the tracer can be laoded
+    // If Runtime metrics are enabled, we can encounter a CannotUnloadAppDomainException
+    // certainly because we are initializing perf counters at that time.
+    // As there are no use case where we would like to load the tracer in that appdomain, just don't
+    if (module_info.assembly.app_domain_name == WStr("NinjectModuleLoader") && !runtime_information_.is_core())
+    {
+        Logger::Warn("JITCompilationStarted: NInjectModuleLoader appdomain deteceted. Not registering startup hook.");
```

Comment:
nit: not sure this needs to be a warning, seems like Info is sufficient?

Reply:
Agreed. We don't want Ninject users to be spammed with this warning in their logs.

Reply:
Fixed in b35d38e53baff3d58f062892e08743051fa81232